app/controllers/backoffice/approval_requests_controller.rb

```diff
@@ -0,0 +1,68 @@
+# frozen_string_literal: true
+
+class Backoffice::ApprovalRequestsController < Backoffice::ApplicationController
+  before_action :find_and_authorize, only: %i[show update]
+
+  def index
+    list_approvals
+  end
+
+  def edit
+  end
+
+  def show
+  end
+
+  def update
+    current_action = params[:approval_request][:current_action]
+    p "EEEEEE #{params[:approval_request][:current_action]}"
+    @approval_request.assign_attributes(
+      permitted_attributes(ApprovalRequest).merge(
+        status: current_action == "requested_for_changes" || current_action.blank? ? :published : :deleted,
+        last_action: current_action
+      )
+    )
+    @approval_request.save
+    @message =
+      Message.new(
+        message: params[:approval_request][:message],
+        author: current_user,
+        author_role: :mediator,
+        scope: :user_direct,
+        messageable: @approval_request
+      )
+    provider = @approval_request.approvable
+    case current_action
+    when "accepted"
+      Provider::Publish.call(provider)
+    when "rejected"
+      Provider::Delete.call(provider)
+    else
+      Provider::Unpublish.call(provider)
+    end
+    list_approvals
+    respond_to do |format|
+      puts "RRRRRRRRRRRR #{@approval_request.valid?} #{@approval_request.errors.messages}"
+      if @approval_request.valid? && Message::Create.new(@message).call
+        notice = _("Message sent successfully")
+        format.turbo_stream { flash.now[:notice] = notice }
+        format.html { redirect_to backoffice_providers_path(notice: notice) }
+      else
+        alert = _("Message not sent")
+        flash.now[:alert] = alert
+        format.json { render :edit, status: :unprocessable_entity }
+        format.html { render :show, status: :unprocessable_entity, alert: alert }
+      end
+    end
+  end
+
+  private
+
+  def find_and_authorize
+    @approval_request = ApprovalRequest.includes(:messages).find(params[:id])
```

Comment:
Do we use those preloaded messages somewhere?

Reply:
Yes, because of listing messages in `show` action and they aren't deleted at all (there're only status changes), so IMHO it's better to preload them in this method.